### DISCUSSION TITLE
Contours in IPF/PF's?

### DISCUSSION THREAD

**Discussion by Mats-Student-Olie** [pyxem/orix#399] [Ideas]
Hi, 

Nice package. Is there any plan to add contour lines to IPF's/ PF's?

Thanks,

Olie

**Comment by hakonanes** [comment]
Hi Olie,

thank you. There are not any plans to add this, no, but I would very much like this feature myself. It is a matter of limited human resources.

Regarding a potential implementation: @harripj made it so that the pole density function (PDF) data can be extracted as a 2D histogram with cartesian (x, y) values for the histogram bin coordinates in the stereographic projection (see [orix.measure.pole_density_function()](https://orix.readthedocs.io/en/stable/reference/generated/orix.measure.pole_density_function.html#orix.measure.pole_density_function)). With these things it should be possible to use contouring in Matplotlib, like shown in [this example](https://matplotlib.org/stable/gallery/images_contours_and_fields/irregulardatagrid.html). I have not tested this, but I believe this is the place to look first.

**Comment by harripj** [comment]
Hi @Mats-Student-Olie,

Sorry for the delay here. I just wrote an example which also demonstrates how to plot contour lines in IPF/PF in another discussion thread.

I'll link it here in case it is of some use to you and for reference: https://github.com/pyxem/orix/discussions/440#discussioncomment-5470704

**Reply by Mats-Student-Olie** [reply]
Ok cool! Thanks!
